--- a/luci-app-cpufreq/luasrc/model/cbi/cpufreq.lua
+++ b/luci-app-cpufreq/luasrc/model/cbi/cpufreq.lua
@@ -55,14 +55,14 @@ for _, policy_num in ipairs(string.split(policy_nums, " ")) do
 	upthreshold.description = translate("Kernel make a decision on whether it should increase the frequency (%)")
 	upthreshold.placeholder = 50
 	upthreshold.default = 50
-	upthreshold:depends("governor", "ondemand")
+	upthreshold:depends("governor" .. policy_num, "ondemand")
 
 	factor = s:taboption(policy_num, Value, "factor" .. policy_num, translate("CPU Switching Sampling rate"))
 	factor.datatype="range(1,100000)"
 	factor.description = translate("The sampling rate determines how frequently the governor checks to tune the CPU (ms)")
 	factor.placeholder = 10
 	factor.default = 10
-	factor:depends("governor", "ondemand")
+	factor:depends("governor" .. policy_num, "ondemand")
 
 end
 
